### ISSUE TITLE
Shrinkwrap file shrinkwrap.yaml not compatible with current pnpm

### ISSUE THREAD

**Issue by plesiecki** [pnpm/pnpm#1437]
### Code to reproduce the issue:
```
#.npmrc
shared-workspace-shrinkwrap=true
link-workspace-packages=true
```
```
#pnpm-workspace.yaml
packages:
  - 'packages/*'
```

+ Single dep inside root and also single workspace package (no deps needed)

### Expected behavior:
🤔

### Actual behavior:
`pnpm i` from root of a workspace ends up with error:
`Shrinkwrap file shrinkwrap.yaml not compatible with current pnpm`

### Additional information:

 - `node -v` prints: 11.0.0
 - Windows, OS X, or Linux?: Ubuntu 18.04.

I noticed combination of `shared-workspace-shrinkwrap` and `link-workspace-packages` causes this error. If only one of those options is applied an error doesn't occur.

**Comment by zkochan** [comment]
yeah, so if you want to use the new `shared-workspace-shrinkwrap` config in your multi-package repo, then you have to remove the `shrinkwrap.yaml` file that was created by an installation without the config.

I know, the error message might be confusing... When you use this new config, the `shrinkwrap.yaml` format is v4 with a new field called `importers`.

**Comment by plesiecki** [comment]
Oh sorry I forgot to mention that it happens on 'clean' repo without any `shrinkwrap.yaml` files

**Comment by zkochan** [comment]
There is not even a `node_modules` folder? There is a shrinkwrap also in `node_modules/.shrinkwrap.yaml`

**Comment by plesiecki** [comment]
Yep, no `node_modules` before and after `pnpm i`. Same with `shrinkwrap.yaml`

**Comment by zkochan** [comment]
ok thanks. I'll try to reproduce

**Comment by plesiecki** [comment]
Should I create repro-repo?

**Comment by zkochan** [comment]
As a temporary workaround you can run `pnpm recursive install` (or `pnpm multi install`/`pnpm m i`).

**Comment by plesiecki** [comment]
Cool. I will give it a try

**Comment by zkochan** [comment]
:ship: 2.17.1

**Comment by zkochan** [comment]
oops, not fixed yet.

**Comment by plesiecki** [comment]
I'm happy to test out pre-release version if you will

**Comment by zkochan** [comment]
fixed in `2.17.2`

**Comment by plesiecki** [comment]
Magnificent work. Thanks. Peace!
